consumer.go

```diff
@@ -161,7 +165,11 @@ func (c *Consumer) ScanShard(
 
 	c.logger.Log("scanning", shardID, lastSeqNum)
 
-	// scan pages of shard
+	return c.scanPagesOfShard(ctx, shardID, lastSeqNum, shardIterator, fn)
+}
+
+func (c *Consumer) scanPagesOfShard(ctx context.Context, shardID, lastSeqNum string, shardIterator *string, fn func(*Record) ScanStatus) error {
+	var isScanStopped bool
```

Comment:
can we remove this var? and use `:=` in the loop:

```go
isScanStopped, err := c.handleRecord(shardID, r, fn)
```

Reply:
@harlow If we do that, the `isScanStopped` variable is going to be created for each record. Is it okay?

Reply:
Yes, I think I prefer it in the loop

Reply:
Okay. Can I know the reason? @harlow

Reply:
Mainly a scoping issue; that boolean doesn't apply to the rest of the function and is initialized pretty far from where it's used.

Reply:
I see. Thank you for the explanation!